main/src/cgeo/geocaching/Geocache.java

```diff
@@ -1000,32 +1000,37 @@ public void setInventory(final List<Trackable> newInventory) {
      *
      * @param newTrackables to be added to the Geocache
      */
-    public void mergeInventory(final List<Trackable> newTrackables) {
-        if (inventory != null) {
-            // Remove non existing Trackable from the same Brand
-            final Iterator<Trackable> iterator = inventory.iterator();
-            while (iterator.hasNext()) {
-                final Trackable trackable = iterator.next();
-                boolean removeTrackable = true;
-                boolean hasBrand = false;
-                for (final Trackable newTrackable : newTrackables) {
-                    if (trackable.getBrand() == newTrackable.getBrand()) {
-                        hasBrand = true;
-                        if (trackable.getUniqueID().equals(newTrackable.getUniqueID())) {
-                            removeTrackable = false;
-                            break;
-                        }
+    public void mergeInventory(final List<Trackable> newTrackables, final List<TrackableBrand> processedBrands) {
+        if (inventory == null) {
+            inventory = new ArrayList<>();
+        }
+
+        // Remove non existing Trackable from the same Brand
+        final Iterator<Trackable> iterator = inventory.iterator();
+        while (iterator.hasNext()) {
+            final Trackable trackable = iterator.next();
+            boolean removeTrackable = true;
+            boolean hasBrand = false;
+            for (final Trackable newTrackable : newTrackables) {
+                if (trackable.getBrand() == newTrackable.getBrand()) {
+                    hasBrand = true;
+                    if (trackable.getUniqueID().equals(newTrackable.getUniqueID())) {
+                        removeTrackable = false;
+                        break;
```

Comment:
Would not it be simpler to just start from a copy of `newTrackables`, and all all items in `inventory` whose brand is not in `processedBrands` (which could be made a set)?